core/src/main/java/org/mule/context/notification/DefaultFlowStackElement.java

```diff
@@ -8,39 +8,50 @@
 
 import org.mule.api.context.notification.FlowStackElement;
 
-import java.io.Serializable;
-import java.util.ArrayList;
-import java.util.List;
-
-public class DefaultFlowStackElement implements FlowStackElement, Serializable
+public class DefaultFlowStackElement implements FlowStackElement
 {
-    private static final long serialVersionUID = -2950239332824176742L;
+    private static final long serialVersionUID = -851491195125245390L;
 
     private String flowName;
-    private List<String> processorPaths;
+    private String currentProcessorPath;
 
     public DefaultFlowStackElement(String flowName)
     {
         this.flowName = flowName;
-        this.processorPaths = new ArrayList<>();
     }
 
-    @Override
+    /**
+     * Keeps the passed processor path as the one being currently executed for the flow backed by this element.
+     * 
+     * @param processorPath the path of the currently executing processor for this flow.
+     */
     public void addInvokedMessageProcessor(String processorPath)
```

Comment:
This method used to add the path to the list of executed processors. Now it only sets the current path. Method name must be updated